spec/20-binary-format.md

```diff
@@ -0,0 +1,106 @@
+# Binary format
+
+Binary format document describes how to encode each field - `traceparent` and
+`tracestate`. The binary format should be used to encode the values of these
+fields. This specification does not specify how these fields should be stored
+and send as a part of a binary payload. The basic implementation may serialize
+those as size of the field followed by the value.
+
+Specification operates with bytes - unsigned 8-bit integer values
+representing values from `0` to `255`. Byte representation as a set of
+bits (big or little endian) MUST be defined by underlying platform and
+out of scope of this specification.
+
+## `Traceparent` binary format
+
+The field `traceparent` encodes the version of the protocol and fields
+`trace-id`, `parent-id` and `trace-flags`. Each field starts with the one byte
+field identifier with the field value following immediately after it. Field
+identifiers are used as markers for additional verification of the value
+consistency and may be used in future for the versioning of the `traceparent`
+field.
+
+``` abnf
+traceparent     = version version_format  
+version         = 1BYTE                   ; version is 0 in the current spec
+version_format  = "{ 0x0 }" trace-id "{ 0x1 }" parent-id "{ 0x2 }" trace-flags
+trace-id        = 16BYTES
+parent-id       = 8BYTES
+trace-flags     = 1BYTE  ; only the least significant bit is used
+```
+
+Unknown field identifier (anything beyond `0`, `1` and `2`) should be treated as
+invalid `traceparent`. All zeroes in `trace-id` and `parent-id` invalidates the
+`traceparent` as well.
+
+## Serialization of `traceparent`
+
+Implementation MUST serialize fields into the field ordering sequence.
+In other words, `trace-id` field should be serialized first, `parent-id`
+second and `trace-flags` - third.
+
+Field identifiers should be treated as unsigned byte numbers and should be
+encoded in big-endian bit order.
+
+Fields `trace-id` and `parent-id` are defined as a byte arrays, NOT a
+long numbers. First element of an array MUST be copied first. When array is
+represented as a memory block of 16 bytes - serialization of `trace-id`
+would be identical to `memcpy` method call on that memory block. This
```

Comment:
> First element of an array MUST be copied first. When array is represented as a memory block of 16 bytes - serialization of `trace-id` would be identical to `memcpy` method call on that memory block.

could you explain what issues you're trying to prevent with these two sentences? They don't seem  to add anything new to just saying "byte arrays" (and array data structure is by definition ordered).

Reply:
Somehow I imagine people trying to cast this memory block to the `int` and getting reversed values. This is why I say "when it's represented as memory block".